src/main/groovy/edu/oregonstate/mist/slashcourses/resources/SlashCourseResource.groovy

```diff
@@ -74,16 +77,75 @@ class SlashCourseResource extends Resource {
     }
 
     /**
-     * Responds to POST requests by reading and returning a message.
+     * Respond to DELETE requests and removes a course object according to course CRN.
      *
-     * @param message
-     * @return message
+     * @param crn
+     * @return response containing the result or error message
      */
-//    @POST
-//    @Consumes(MediaType.TEXT_PLAIN)
-//    @Produces(MediaType.TEXT_PLAIN)
-//    public Response postMessage(String message, @Auth AuthenticatedUser authenticatedUser) {
-//        ResponseBuilder responseBuilder = ok(message)
-//        responseBuilder.build()
-//    }
+    @Path('{crn: \\d+}')
+    @DELETE
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteByCRN (@PathParam('crn') Integer crn) {
+        slashCourseDAO.deleteByCRN(crn)
+        Response.ok().build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postCourse (@Valid SlashCourse newCourse) {
+
+        Response returnResponse
+        try {
+
+            List<Integer> crnList = slashCourseDAO.getAllCRN()
+
+            if (newCourse.getInstructor() != null && !crnList.contains(newCourse.getCrn())) {
```

Comment:
@floppyears I think I found the problem I had this morning. I guess the reason is because `!crnList.contains(newCourse.getCrn())` is not equal to `!crnList[newCourse.crn]`. What I wanted to do is to check if the CRN is in my CRN list or not, rather than use CRN as an index of the list :)